test/Transaction.purs

```diff
@@ -0,0 +1,45 @@
+module Test.Transaction (suite) where
+
+import Prelude
+
+import Data.BigInt as BigInt
+import Data.Either (Either(Left, Right))
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Newtype (unwrap)
+import Effect.Aff (Aff)
+import Effect.Class (liftEffect)
+import Effect.Exception (throw)
+import Deserialization.PlutusData as Deserialization.PlutusData
+import Serialization.WitnessSet as Serialization.WitnessSet
+import Mote (group, test)
+import Test.Spec.Assertions (shouldEqual)
+import TestM (TestPlanM)
+import Transaction (attachDatum)
+import Types.Transaction
+  ( Transaction(Transaction)
+  , TransactionWitnessSet(TransactionWitnessSet)
+  )
+import Types.PlutusData (Datum(Datum), PlutusData(Integer))
+
+suite :: TestPlanM Unit
+suite = group "attach datums to tx"
+  $ test "datum should be correctly attached" testAttachDatum
+
+testAttachDatum :: Aff Unit
+testAttachDatum = liftEffect $
```

Comment:
We could use a test which shows that all witnesses are preserved

Reply:
Good idea